Src/Coravel/Scheduling/Schedule/Event/ScheduledEvent.cs

```diff
@@ -216,14 +216,19 @@ public IScheduledEventConfiguration PreventOverlapping(string uniqueIdentifier)
         public IScheduledEventConfiguration When(Task<bool> task)
```

Comment:
I think just using `Func<Task<bool>> task` would be the appropriate public api 👍